lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -801,10 +801,6 @@ class TypecastOpConversionPattern
 
     DataType outputDataType = outputLayoutAttr.getDataType();
 
-    if (op->getUsers().empty()) {
```

Comment:
Unrelated to this PR, we should have a way to optimize out cases like this when op has no users. I am not sure if canonicalization will do the job here if the op has memory effects. Can you please discuss this case with @azecevicTT and whoever else is working on folding/canonicalization in dialects team?

Reply:
@mrakitaTT This should be covered by `remove-dead-values` pass that's run as the part of `ttir-to-ttnn-backend-pipeline`. You're right about memory effects, but with `NoMemoryEffects` trait added in https://github.com/tenstorrent/tt-mlir/pull/1481 it should now remove all ops that don't have users. I've raised some concerns regarding that PR, but `typecast` and most other ops should have `Pure` semantics.